src/Notifications/KillMail/AbstractKillMailNotification.php

```diff
@@ -0,0 +1,197 @@
+<?php
+/**
+ * MIT License.
+ *
+ * Copyright (c) 2019. Felix Huber
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+namespace Herpaderpaldent\Seat\SeatNotifications\Notifications\KillMail;
+
+use Herpaderpaldent\Seat\SeatNotifications\Notifications\AbstractNotification;
+use Seat\Eveapi\Models\Killmails\KillmailDetail;
+use Seat\Eveapi\Models\Killmails\KillmailVictimItem;
+use Seat\Eveapi\Models\Market\Price;
+
+abstract class AbstractKillMailNotification extends AbstractNotification
+{
+    /**
+     * @var string
+     */
+    public $killmail_detail;
+
+    /**
+     * @var string
+     */
+    public $image;
+
+    /**
+     * AbstractKillMailNotification constructor.
+     */
+    public function __construct(int $killmail_id)
+    {
+
+        parent::__construct();
+
+        $this->killmail_detail = KillmailDetail::find($killmail_id);
+        $this->image = sprintf('https://imageserver.eveonline.com/Type/%d_64.png',
+            $this->killmail_detail->victims->ship_type_id);
+
+        array_push($this->tags, 'killmail_id:' . $killmail_id);
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getTitle(): string
+    {
+
+        return 'Kill Mail Notification';
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getDescription(): string
+    {
+
+        return 'Receive a notification about new kill mails.';
+    }
+
+    /**
+     * @return bool
+     */
+    final public static function isPublic(): bool
+    {
+
+        return true;
+    }
+
+    /**
+     * @return bool
+     */
+    final public static function isPersonal(): bool
+    {
+
+        return false;
```

Comment:
This overload is unneeded as it's default value 😜

Reply:
I can't confirm this. Doesn't contain the default. https://github.com/herpaderpaldent/seat-notifications/blob/24a1c7b99eb07fa4cccaab6bec863692342c7328/src/Notifications/KillMail/AbstractKillMailNotification.php also i believe it is the right place to define which notifications are present